lib/ansible/executor/powershell/module_manifest.py

```diff
@@ -104,9 +106,11 @@ def scan_module(self, module_data, fqn=None, wrapper=False, powershell=True):
                         # tolerate windows line endings by stripping any remaining
                         # newline chars
                         module_util_name = to_text(match.group(1).rstrip())
+                        match_dict = match.groupdict()
+                        optional = match_dict.get('optional', None) is not None
 
                         if module_util_name not in check[1].keys():
-                            module_utils.add((module_util_name, check[2], fqn))
+                            module_utils.add((module_util_name, check[2], fqn, optional))
```

Comment:
I wish this were a Dataclass or namedtuple, but doesn't need to hold up the works.